float64 rotation
float64[] com1
float64[] com2